src/ai/backend/manager/models/endpoint.py

```diff
@@ -631,6 +645,86 @@ class Meta:
     items = graphene.List(Endpoint, required=True)
 
 
+class ModifyEndpointInput(graphene.InputObjectType):
+    resource_slots = graphene.JSONString()
+    resource_opts = graphene.JSONString()
+    cluster_mode = graphene.String()
+    cluster_size = graphene.Int()
+    desired_session_count = graphene.Int()
+    image = ImageRefType()
+    name = graphene.String()
+    resource_group = graphene.String()
+    open_to_public = graphene.Boolean()
+
+
+class ModifyEndpoint(graphene.Mutation):
+    class Arguments:
+        endpoint_id = graphene.UUID(required=True)
+        props = ModifyEndpointInput(required=True)
+
+    ok = graphene.Boolean()
+    msg = graphene.String()
+    endpoint = graphene.Field(lambda: Endpoint, required=False)
```

Comment:
Please add a description to introducing fileds for version compatibility like below:
https://github.com/lablup/backend.ai/blob/d8ac38b4752e40d37444438c54e0ddbb88875038/src/ai/backend/manager/models/group.py#L261

Reply:
@yomybaby I think field `endpoint` should be applied in 23.09.9, rather than 24.03.0, since the milestone of this feature is 23.09. I'll apply it right away.